### PR TITLE
Issue 18567 add cors to graphql servlet

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.HashMap;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.dotcms.util.IntegrationTestInitService;
+
+public class DotGraphQLHttpServletTest {
+
+    
+    @BeforeClass
+    public static void prepare() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:13]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void testing_cors_headers() {
+        
+        HashMap<String,String> headers = new DotGraphQLHttpServlet().corsHeaders.apply();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:21]: Codacy Issue found: [Local variable 'headers' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * See an example here: {@link MockHttpRequest#MockHttpRequest(String, String)}
+ */
+public class MockHeaderResponse extends HttpServletResponseWrapper implements MockResponse {
+  final Map<String, String> headers = new HashMap<String, String>();
+
+  public MockHeaderResponse(HttpServletResponse response) {
+    super(response);
+    if (response.getHeaderNames() != null) {
+      Collection<String> oldHeaders = response.getHeaderNames();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:23]: Codacy Issue found: [Local variable 'oldHeaders' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package com.dotcms.graphql;
+
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:3]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        
+        // this property is specifically overriden for graphql
+        assertEquals(headers.get("access-control-expose-headers"), "Content-Type,Cache-Control");
+    }
+
+    @Test
+    public void testing_GETRequestToGraphQLServer_returnResponseWithExpectedHeaders()
+            throws ServletException, IOException {
+
+        MockHttpRequest request = new MockHttpRequest("localhost", "/");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:44]: Codacy Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        MockHeaderResponse response = new MockHeaderResponse(new MockHttpResponse());
+        DotGraphQLHttpServlet graphQLHttpServlet = new DotGraphQLHttpServlet();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:46]: Codacy Issue found: [Local variable 'graphQLHttpServlet' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  @Override
+  public Collection<String> getHeaderNames() {
+    return headers.keySet();
+  }
+
+  public void setHeader(final String name, final String o) {
+    headers.put(name, o);
+  }
+
+  public void addHeader(final String name, final String o) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:53]: Codacy Issue found: [Avoid variables with short names like o](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Mock {@link HttpServletRequest} with a {@link Map} as part of the class that will contain the
+ * Headers.
+ *
+ * See an example here: {@link MockHttpRequest#MockHttpRequest(String, String)}
+ */
+public class MockHeaderResponse extends HttpServletResponseWrapper implements MockResponse {
+  final Map<String, String> headers = new HashMap<String, String>();
+
+  public MockHeaderResponse(HttpServletResponse response) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:20]: Codacy Issue found: [Parameter 'response' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  public void setHeader(final String name, final String o) {
+    headers.put(name, o);
+  }
+
+  public void addHeader(final String name, final String o) {
+    headers.put(name, o);
+  }
+
+  @Override
+  public Collection<String> getHeaders(final String name) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:58]: Codacy Issue found: [Overriding method merely calls super](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    super(response);
+    if (response.getHeaderNames() != null) {
+      Collection<String> oldHeaders = response.getHeaderNames();
+      oldHeaders.forEach(header->{
+        this.headers.put(header, response.getHeader(header));
+      });
+    }
+  }
+
+  public MockHeaderResponse(HttpServletResponse response, final String key, final String value) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:30]: Codacy Issue found: [Parameter 'response' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+import static org.junit.Assert.*;
+
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.response.MockHeaderResponse;
+import com.dotcms.mock.response.MockHttpResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:12]: Codacy Issue found: [Avoid unused imports such as 'javax.servlet.http.HttpServletResponse'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.dotcms.util.IntegrationTestInitService;
+
+public class DotGraphQLHttpServletTest {
+
+    
+    @BeforeClass
+    public static void prepare() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:21]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void testing_cors_headers() {
+        
+        HashMap<String,String> headers = new DotGraphQLHttpServlet().corsHeaders.apply();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:29]: Codacy Issue found: [Local variable 'headers' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+    this(response);
+    headers.put(key, value);
+  }
+
+  public HttpServletResponse response() {
+    return this;
+  }
+
+  @Override
+  public String getHeader(String name) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:40]: Codacy Issue found: [Parameter 'name' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+import static org.junit.Assert.*;
+
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.response.MockHeaderResponse;
+import com.dotcms.mock.response.MockHttpResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:11]: Codacy Issue found: [Avoid unused imports such as 'javax.servlet.http.HttpServletRequest'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+    return headers.get(name);
+  }
+
+  @Override
+  public Collection<String> getHeaderNames() {
+    return headers.keySet();
+  }
+
+  public void setHeader(final String name, final String o) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:49]: Codacy Issue found: [Avoid variables with short names like o](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+        // this property is specifically overriden for graphql
+        assertEquals(headers.get("access-control-expose-headers"), "Content-Type,Cache-Control");
+    }
+
+    @Test
+    public void testing_GETRequestToGraphQLServer_returnResponseWithExpectedHeaders()
+            throws ServletException, IOException {
+
+        MockHttpRequest request = new MockHttpRequest("localhost", "/");
+        MockHeaderResponse response = new MockHeaderResponse(new MockHttpResponse());
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:45]: Codacy Issue found: [Local variable 'response' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 9 above)
Comment (wezell) [reply dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:58]: I guess we can update this when we need it for a test.

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:20]: Codacy Issue found: [Parameter 'response' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:44]: Codacy Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:40]: Codacy Issue found: [Parameter 'name' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:30]: Codacy Issue found: [Parameter 'response' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:3]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:53]: Codacy Issue found: [Avoid variables with short names like o](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:29]: Codacy Issue found: [Local variable 'headers' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:46]: Codacy Issue found: [Local variable 'graphQLHttpServlet' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 17 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:45]: Codacy Issue found: [Local variable 'response' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:12]: Codacy Issue found: [Avoid unused imports such as 'javax.servlet.http.HttpServletResponse'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:23]: Codacy Issue found: [Local variable 'oldHeaders' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:58]: Codacy Issue found: [Overriding method merely calls super](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:11]: Codacy Issue found: [Avoid unused imports such as 'javax.servlet.http.HttpServletRequest'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/DotGraphQLHttpServletTest.java:21]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

File: dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java:49]: Codacy Issue found: [Avoid variables with short names like o](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**